### ISSUE TITLE
[Q] Using Sweeps with SB3 - All runs in Sweeps drop to <null>

### ISSUE THREAD

**Issue by melzwail** [wandb/wandb#6217]
I am using Sweeps from W&B to optimize the hyperparameters in an RL agent built using SB3. As a target for the optimization process, I'm aiming to maximize the reward. When building the config dict for Sweeps, what should I have in the "metric" entry for the "name" key? Is it "reward", "ep_rew_mean", "rollout/ep_rew_mean" or something else? 
I played with both "ep_rew_mean" and "rollout/ep_rew_mean", In both cases I have some graphical output in my dashbourd for the Sweep dashboard but in both cases and for all 10 sweeps the values of ep_rew_mean go down to <null>. The curves change in each run taking different values of "learning rate" and "ent_coef" but they all drop to <null> at the right end at the "ep_rew_mean" axis.
Here's the related snippet of the code:
```
    sweep_config = {
        'name': 'my-sweep',
        'method': 'bayes',  
        'metric': {
            'name': 'ep_rew_mean', 
            'goal': 'maximize'},
        'parameters': {
            'learning_rate': {                          
                'distribution': 'uniform', 
                'min': 1e-6,
                'max': 1e-2},
            'batch_size': {'value': 50},
            'gamma': {'value': 0.995},  
            'n_steps': {'value': 100},
            'ent_coef': {'values': [0.001, 0.0001]},
            'policy_type': {'value': "MlpPolicy"},  
            'total_timesteps': {'value': 10100} 
        }
    }

    sweep_id = wandb.sweep(sweep=sweep_config, project='wf1', entity='mmzwail')

    def train(config=None):
        with wandb.init(config=config):
            config = wandb.config

            agent = PPO(
                policy=config.policy_type,
                env=env,
                n_steps=config.n_steps,
                learning_rate=config.learning_rate,
                verbose=1,
                gamma=config.gamma,
                batch_size=config.batch_size,
                ent_coef=config.ent_coef)
            )

            agent.learn(
                total_timesteps=config.total_timesteps,
                reset_num_timesteps=False,
                progress_bar=True,
                 callback=WandbCallback(
                gradient_save_freq=100,
                model_save_path=f"models/{wandb.run.id}",
                verbose=2,
                )
            )
wandb.agent(sweep_id, train, count=10)

```

**Comment by melzwail** [comment]
Hi @luisbergua, thank you very much for your quick response. I think I figured this issue out. My Sweeps are working now (not totally done) with reasonable results so far.
The issue originated from some kind of hacky workaround to a previous issue. Before, optimizing my hyperparameters with sweeps, I was running regular W&B logging and I was getting some kind of error ( I forgot what it was). So I searched for that issue on this platform, turns out a lot of people were facing the same issue and the solution was to omit the line re tensorboard logging in wandb.init(), the 'sync_tensorboard=True" line and also the one re tensorboard when defining my RL agent, and instead add this line: `wandb.tensorboard.patch(save=False)`. 
This workaround solved that issue till I started using Sweeps. All my sweeps were missing the loss/performance graphs, while the Sweeps curves were all dropping to <null> at the end.
So, I just walked back through all the changes in the code and the one that fixed this new issue is to to uncomment the tensorboard logging arg when creating the RL agent instance. 
Here's the working code snippet:
`            agent = PPO(
                policy=config.policy_type,
                # env=env,
                env=local_env,  # Use the local environment here
                n_steps=config.n_steps,
                learning_rate=config.learning_rate,
                verbose=1,
                gamma=config.gamma,
                batch_size=config.batch_size,
                ent_coef=config.ent_coef,
                tensorboard_log=str(log_dir)
            )

            best_model_callback = SaveBestModel(check_freq=100, model_dir=str(model_dir))

            agent.learn(
                total_timesteps=config.total_timesteps,
                reset_num_timesteps=False,
                progress_bar=True,
                callback=[WandbCallback(
                    gradient_save_freq=100,
                    model_save_path=f"models/{wandb.run.id}",
                    verbose=2,
                ), best_model_callback]
                    
            )

wandb.agent(sweep_id, train, count=10)
`

**Comment by luisbergua** [comment]
Hi @melzwail, thank you so much for sharing such a detailed explanation on this and great to hear you figured out the solution!

**Comment by sydholl** [comment]
WandB Internal User commented: 
melzwail commented: 
Hi @luisbergua, thank you very much for your quick response. I think I figured this issue out. My Sweeps are working now (not totally done) with reasonable results so far.
The issue originated from some kind of hacky workaround to a previous issue. Before, optimizing my hyperparameters with sweeps, I was running regular W&B logging and I was getting some kind of error ( I forgot what it was). So I searched for that issue on this platform, turns out a lot of people were facing the same issue and the solution was to omit the line re tensorboard logging in wandb.init(), the 'sync_tensorboard=True" line and also the one re tensorboard when defining my RL agent, and instead add this line: `wandb.tensorboard.patch(save=False)`. 
This workaround solved that issue till I started using Sweeps. All my sweeps were missing the loss/performance graphs, while the Sweeps curves were all dropping to <null> at the end.
So, I just walked back through all the changes in the code and the one that fixed this new issue is to to uncomment the tensorboard logging arg when creating the RL agent instance. 
Here's the working code snippet:
`            agent = PPO(
                policy=config.policy_type,
                # env=env,
                env=local_env,  # Use the local environment here
                n_steps=config.n_steps,
                learning_rate=config.learning_rate,
                verbose=1,
                gamma=config.gamma,
                batch_size=config.batch_size,
                ent_coef=config.ent_coef,
                tensorboard_log=str(log_dir)
            )

            best_model_callback = SaveBestModel(check_freq=100, model_dir=str(model_dir))

            agent.learn(
                total_timesteps=config.total_timesteps,
                reset_num_timesteps=False,
                progress_bar=True,
                callback=[WandbCallback(
                    gradient_save_freq=100,
                    model_save_path=f"models/{wandb.run.id}",
                    verbose=2,
                ), best_model_callback]
                    
            )

wandb.agent(sweep_id, train, count=10)
`

**Comment by sydholl** [comment]
WandB Internal User commented: 
luisbergua commented: 
Hi @melzwail, thank you so much for sharing such a detailed explanation on this and great to hear you figured out the solution!
